Bio/SeqUtils/__init__.py

```diff
@@ -577,6 +578,131 @@ def six_frame_translations(seq, genetic_code=1):
     return res
 
 
+class CodonAdaptationIndex(dict):
+    """A codon adaptation index (CAI) implementation.
+
+    Implements the codon adaptation index (CAI) described by Sharp and
+    Li (Nucleic Acids Res. 1987 Feb 11;15(3):1281-95).
+
+    NOTE - This implementation does not currently cope with alternative genetic
+    codes: only the synonymous codons in the standard table are considered.
+    """
+
+    # each tuple consists of codons encoding the same amino acid
+    synonymous_codons = (
```

Comment:
Feel like we could pull this from ``Bio/Data/CodonTable.py`` but not essential.

Reply:
done